tests/handlers/test_appservice.py

```diff
@@ -412,6 +413,83 @@ def prepare(self, reactor, clock, hs):
             "exclusive_as_user", "password", self.exclusive_as_user_device_id
         )
 
+    def _notify_interested_services(self):
+        # This is normally set in `notify_interested_services` but we need to call the
+        # internal async version so the reactor gets pushed to completion.
+        self.hs.get_application_service_handler().current_max += 1
+        self.get_success(
+            self.hs.get_application_service_handler()._notify_interested_services(
+                RoomStreamToken(
+                    None, self.hs.get_application_service_handler().current_max
+                )
+            )
+        )
+
+    @parameterized.expand(["@local_as_user:test", "@interesting_user:remote"])
+    def test_match_interesting_room_members(self, interesting_user):
+        """
+        Test to make sure that a interesting user (local or remote) in the room is notified
+        when someone else in the room sends a message.
+        """
```

Comment:
TODO: Potentially merge these tests into https://github.com/matrix-org/synapse/pull/13958

Per-discussion, https://github.com/matrix-org/synapse/pull/14000#issuecomment-1266552433, they need to be adapted to not expect a remote user anymore though

Reply:
Adapted and moved over to https://github.com/matrix-org/synapse/pull/13958 ✅